docs/_sources/week02/display-reports/index.rst.txt

```diff
@@ -0,0 +1,158 @@
+
+Create ReportController.java
+----------------------------
+
+We have a Report class, a report table with records, and a ReportRepository to manage the flow of information between Spring, and the database. Now we need some way of handling an HTTP request from OpenLayers, and the response needs to include a GeoJSON representation of our report data.
+
+As we have learned from this class the ``@Controller`` annotation allows us to handle HTTP requests, and serve up HTTP responses. We will need a controller, an endpoint, and some logic to get the information from the database, turn it into GeoJSON, and then package it into an HTTP response.
+
+Again, as a good practice we should store all of our controlles, in their own directory called ``controllers/``. Create this new directory, and add a new file called ``ReportController.java``.
+
+In this file you will need to setup a new method handler for an endpoint that will return a GeoJSON representation of our report objects.
+
+Convert Java Object to GeoJSON
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Look over the ``AirportController.java`` file from the `Airwaze code base <https://gitlab.com/LaunchCodeTraining/airwaze-studio/blob/master/src/main/java/com/launchcode/gisdevops/controllers/AirportController.java>`_.
+
+You will note in this file the ``getAirports()`` request handler method is returning a FeatureCollection object. The FeatureCollection is the GeoJSON! Look over both the `FeatureCollection.java <https://gitlab.com/LaunchCodeTraining/airwaze-studio/blob/master/src/main/java/com/launchcode/gisdevops/features/FeatureCollection.java>`_ and `Feature.java <https://gitlab.com/LaunchCodeTraining/airwaze-studio/blob/master/src/main/java/com/launchcode/gisdevops/features/Feature.java>`_ files.
+
+You will need to recreate both of these files in your project in order to return our report data as GeoJSON. A Featue is one record in GeoJSON. A FeatureCollection is multiple Features together, still in GeoJSON.
+
+Notice that the Feature, and FeatureCollection classes are using a new annotation we haven't seen yet: ``@JsonSerialize(using = GeometrySerializer.class)``. The JsonSerializer takes our object and converts it into JSON. To convert our GeoINT information we will also need ``GeoJSONSerializer.java``, and ``WktHelper.java``.
+
+All four of these files are how we are converting our Java Report objects into GeoJSON that is usable by JavaScript.
+
+In the handler method you will need to load all of youre reports from the ReportRepository, loop through each report, create a new Feature from each report, and add each Feature to the FeatureCollection, and finally return the FeatureCollection as a part of the HTTP response.
+
+Create ReportControllerTests.java
+---------------------------------
+
+Since we are creating a new class we also need to test the class. Look at the ``AirportControllerTest.java`` file from the `airwaze studio <../../studios/airwaze/>`_ for help.
+
+Amend script.js
```

Comment:
The guidance in this section seems a bit light, but I can't say for sure since I'm not new to the project.

Reply:
Similarly to the ReportTest.java file. I advised students to create a specific method, and what should go in that method, but don't provide any code. They will have to figure that out on their own by looking at previous examples, talking to other students, or working with an instructor.

Reply:
I also added a section for ReportRepositoryTests.java. All three testing sections contain a hint on what method to create, and what should be included in the testing method.